### PR TITLE
Refactor: use the verifying client in the hammer

### PR DESCRIPTION
**Pull request by mhutchinson**
The hammer was duplicating logic in the verifying client. It makes sense to reuse this logic.

This addresses some of the issues brought up investigating #1845

### CODE REVIEW

File: testonly/hammer/hammer.go [hunk 1]
```diff
@@ -568,62 +567,23 @@ func (s *hammerState) doGetLeaves(ctx context.Context, prng *rand.Rand, latest b
...
-		return fmt.Errorf("failed to verify root: %v", err)
-	}
-	for i, inc := range rsp.MapLeafInclusion {
-		if err := s.verifier.VerifyMapLeafInclusionHash(root.RootHash, inc); err != nil {
-			return fmt.Errorf("failed to verify inclusion proof for Index=%x (%d/%d) in revision %d of tree %d: %v", inc.Leaf.Index, i, len(rsp.MapLeafInclusion), root.Revision, s.cfg.MapID, err)
+			return fmt.Errorf("failed to GetAndVerifyMapLeavesByRevision: %v", err)
 		}
 	}
-
-	if err := contents.CheckContents(rsp.MapLeafInclusion, s.cfg.ExtraSize); err != nil {
```
Comment (gdbelvin) [inline testonly/hammer/hammer.go]: Ad a TODO to check whether the contents match the expected values?

File: testonly/hammer/hammer.go [hunk 2]
```diff
@@ -780,50 +732,38 @@ func (s *hammerState) setLeavesInvalid(ctx context.Context, prng *rand.Rand) err
...
-	root, err := s.verifier.VerifySignedMapRoot(rsp.MapRoot)
+	root, err := s.vc.VerifySignedMapRoot(rsp.MapRoot)
 	if err != nil {
 		return err
 	}
 	glog.V(2).Infof("%d: got SMR(time=%q, rev=%d)", s.cfg.MapID, time.Unix(0, int64(root.TimestampNanos)), root.Revision)
 
-	if !proto.Equal(rsp.MapRoot, smr) {
-		return fmt.Errorf("get-smr-rev(@%d)=%+v, want %+v", rev, rsp.MapRoot, smr)
+	if !bytes.Equal(root.RootHash, smrRoot.RootHash) {
```
Comment (gdbelvin) [inline testonly/hammer/hammer.go]: If you'd like to check a struct that *contains* a proto, the right way to do this is to use the `cmp` library.

(From memory)
```
cmp.Equal(A,B, cmpopt.Equals(proto.Equal))
```

File: testonly/hammer/hammer.go (hunk 1 above)
Comment (mhutchinson) [reply testonly/hammer/hammer.go]: Added this code back in. Got a bit aggressive with the pruning.

Comment (gdbelvin) [reply testonly/hammer/hammer.go]: Awaiting a github push...

File: testonly/hammer/hammer.go (hunk 2 above)
Comment (mhutchinson) [reply testonly/hammer/hammer.go]: No protos involved here. I'm using DeepEqual now.

File: testonly/hammer/hammer.go (hunk 1 above)
Comment (mhutchinson) [reply testonly/hammer/hammer.go]: Done. Normally I feel like github starts a review session for my comments, but this time it pushed directly without the batch. Anyway, race condition has resolved, and this is now addressed.

**Comment by mhutchinson** [comment]
I also took the opportunity to rebase on master so I could verify locally that these changes were working, which require #1849 to work.

**Pull request merged**